lib/fog/azurerm/models/storage/file.rb

```diff
@@ -4,147 +4,386 @@ class AzureRM
+          merge_attributes(data)
+
+          true
         end
 
-        def set_properties(properties = {})
+        # Destroy file.
+        #
+        #     required attributes: directory, key
+        #
+        # @param options [Hash]
+        # @option options versionId []
+        #
+        # @return [Boolean] true if successful
+        #
+        def destroy(options = {})
           requires :key
           requires :directory
-          service.set_blob_properties(directory, key, properties)
-          merge_attributes(properties)
+          attributes[:body] = nil
+          service.delete_blob(directory.key, key, options)
+
+          true
         end
 
-        def get_metadata(options = {})
-          requires :key
+        # Get whether the file can be accessed by anonymous.
+        #
+        # @return [Boolean]
+        #
+        def public?
           requires :directory
-          merge_attributes(metadata: service.get_blob_metadata(directory, key, options))
+
+          # TBD: The blob can be accessed if read permision is set in one access policy of the container.
+          directory.acl == 'container' || directory.acl == 'blob'
         end
 
-        def set_metadata(metadata, options = {})
+        # Get publicly accessible url.
+        #
+        #     required attributes: directory, key
+        #
+        # @param options [Hash]
+        # @option options [String] scheme Sets which URL to get, http or https. Options: https or http. Default is https.
+        #
+        # @return [String] A public url.
+        #
+        def public_url(options = {})
+          requires :directory, :key
+          options[:scheme] == 'https' if options[:scheme].nil?
+          @service.get_blob_url(directory.key, key, options) if public?
+        end
+
+        # Get a url for file.
+        #
+        #     required attributes: key
+        #
+        # @param expires [Time] The time at which the shared access signature becomes invalid, in a UTC format.
+        # @param options [Hash]
+        # @option options [String] scheme Sets which URL to get, http or https. Options: https or http. Default is https.
+        #
+        # @return [String] A public url which will expire after the specified time.
+        #
+        def url(expires, options = {})
           requires :key
-          requires :directory
-          service.set_blob_metadata(directory, key, metadata, options)
-          merge_attributes(metadata: metadata)
+          collection.get_url(key, expires, options)
         end
 
-        def self.parse(blob)
-          if (blob.is_a? Hash) || (blob.is_a? Array)
-            parse_hash blob
-          else
-            parse_object blob
+        private
+
+        # Wait the copy operation to finish
+        def wait_copy_operation_to_finish(target_directory_key, target_file_key, copy_id, copy_status, timeout)
+          start_time = Time.new
+          while copy_status == 'pending'
```

Comment:
Must we should do this? Only this function will use these status. And we can see lots of constants in different files in this repo. And other cloud providers also do. I think we should define constants if they are used in cross modules. How do you think?